### PR TITLE
docs: Replace some mentions of `jj co` with `jj new`.

### LINKED ISSUE #3487: docs: still referring to `jj co`
## Description
branches.md still refers to `jj co` in two places: first paragraph and in the section on conflicts.

git-comparison.md confusingly says
> For example, if you check out a commit with a branch, new commits are created on top of the branch, 

## Specifications
- Version: 0.16.0

### PR DESCRIPTION
**Pull request by PhilipMetzger**
Also move some sentences away from "check out" to "create a new revision ...", even if it isn't the phrasing we want.

Fixes #3487

### CODE REVIEW

File: docs/git-comparison.md [hunk 1]
```diff
@@ -32,9 +32,9 @@ various use cases.
   branch', and new commits will automatically update the branch. This is
   necessary in Git because Git might otherwise lose track of the new commits.
   Jujutsu does not have a 'current branch'; instead, you update branches
-  manually. For example, if you check out a commit with a branch, new commits
-  are created on top of the branch, then you issue a later command to update the
-  branch.
+  manually. For example, if you start work on top of a commit with a branch,
```
Comment (PhilipMetzger) [inline docs/git-comparison.md:35]: cc @martinvonz, does this work for you? As you haven't had a clear idea on how we should replace the "check out" verb.

Comment (martinvonz) [reply docs/git-comparison.md:35]: SGTM

Comment (PhilipMetzger) [reply docs/git-comparison.md:35]: thanks

**Comment by ilyagr** [comment]
Thank you!

**Pull request merged**